src/loaders/ncbi_downloader/ncbi_downloader.py

```diff
@@ -231,9 +288,26 @@ def download_genome_files(gene_ids: list[str], target_file_ext: list[str], exclu
     return failed_ids
 
 
+def create_softlink_between_csd_and_work_dir(
+        csd: str,
+        work_dir: str,
+        genome_ids_unprocssed: list[str],
+        failed_ids: list[str] = []
+) -> None:
+    genome_ids = set(genome_ids_unprocssed) - set(failed_ids)
+    for genome_id in genome_ids:
+        genome_dir = os.path.join(work_dir, genome_id)
+        csd_genome_dir = os.path.join(csd, genome_id)
+        loader_helper.create_softlink(csd_genome_dir, genome_dir)
```

Comment:
We need to create the `csd_genome_dir` and link the files within. Just same way as WS downloader. Otherwise, it will break the tool runner which assumes this is a directory. 
https://github.com/kbase/collections/blob/main/src/loaders/compute_tools/tool_common.py#L195-L196

Reply:
NV. I looked again, I think WS downloader and NCBI downloader does behaviors the same. I will need to fix the tool runner.

Reply:
It is indeed a directory after softlink. I am not sure why you still need to fix the tool runner for that.

Reply:
hmmmm. I am seeing those are all links not directories. 

```
(base) tgu@login23 2023.06  $ pwd
/global/u2/t/tgu/collectionssource/CI/GROW/2023.06
(base) tgu@login23 2023.06  $ file -b *
symbolic link to /global/homes/t/tgu/sourcedata/WS/CI/69037/69037_10_1
symbolic link to /global/homes/t/tgu/sourcedata/WS/CI/69037/69037_2_1
symbolic link to /global/homes/t/tgu/sourcedata/WS/CI/69037/69037_4_1
symbolic link to /global/homes/t/tgu/sourcedata/WS/CI/69037/69037_6_1
symbolic link to /global/homes/t/tgu/sourcedata/WS/CI/69037/69037_8_1
```

Reply:
where do you get those? Do they exist in /global?

Reply:
I download those in using the WS downloader. The full GROW set is in `/global/cfs/cdirs/kbase/collections/collectionssource/CI/GROW` 

```
(base) tgu@login23 GROW  $ pwd
/global/cfs/cdirs/kbase/collections/collectionssource/CI/GROW
(base) tgu@login23 GROW  $ ls
2023.06
(base) tgu@login23 GROW  $ file -b *
setgid, directory
(base) tgu@login23 GROW  $ cd 2023.06/
(base) tgu@login23 2023.06  $ file -b *
symbolic link to /global/cfs/cdirs/kbase/collections/sourcedata/WS/CI/69037/69037_1000_1
symbolic link to /global/cfs/cdirs/kbase/collections/sourcedata/WS/CI/69037/69037_100_1
symbolic link to /global/cfs/cdirs/kbase/collections/sourcedata/WS/CI/69037/69037_1002_1
symbolic link to /global/cfs/cdirs/kbase/collections/sourcedata/WS/CI/69037/69037_1004_1
symbolic link to /global/cfs/cdirs/kbase/collections/sourcedata/WS/CI/69037/69037_1006_1
symbolic link to /global/cfs/cdirs/kbase/collections/sourcedata/WS/CI/69037/69037_1008_1
symbolic link to /global/cfs/cdirs/kbase/collections/sourcedata/WS/CI/69037/69037_10_1
```

Reply:
They are dirs and all linked.

<img width="1356" alt="Screen Shot 2023-06-30 at 12 00 46 PM" src="https://github.com/kbase/collections/assets/22202096/e10b8245-2b74-4ee6-8c79-7575b436bc3c">

Reply:
the `ls -l` shows each genome directory is a link. They are not directories. I lost my clue why `os.path.isdir` works in NERSC but not locally in my container. 

It breaks tool runner when I test locally. But it might work in NERSC? IDK. 
https://github.com/kbase/collections/blob/main/src/loaders/compute_tools/tool_common.py#L195-L196

in NERSC:
```
Python 3.11.4 | packaged by conda-forge | (main, Jun 10 2023, 18:08:17) [GCC 12.2.0]
Type 'copyright', 'credits' or 'license' for more information
IPython 8.12.0 -- An enhanced Interactive Python. Type '?' for help.

In [1]: import os

In [2]: os.path.isdir('/global/cfs/cdirs/kbase/collections/collectionssource/NONE/GTDB/207/GCA_002506415.1')
Out[2]: True

In [3]: os.path.isdir('/global/cfs/cdirs/kbase/collections/collectionssource/NONE/GTDB/207')
Out[3]: True
```

VS in my local container. 
```
Python 3.11.4 | packaged by conda-forge | (main, Jun 10 2023, 18:08:17) [GCC 12.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import os
>>> os.path.isdir('root/collectionssource/NONE/GTDB/207/GCF_000980105.1')
False
>>> os.path.isdir('root/collectionssource/NONE/GTDB/207')
True
```

Reply:
Hi, on my local, it is still dir.

Reply:
NV. I figured. I should just mount the source directory (`sourcedata/NCBI/NONE`). Instead I was mounting the directory (`collectionssource/NONE/GTDB/207`)  with softlinks.